tracing-subscriber/src/filter/targets.rs

```diff
@@ -277,6 +277,31 @@ impl Targets {
         self
     }
 
+    /// Returns the explicitly set default level for this filter, if any.
+    ///
+    /// If a default level was set explicitly using [`with_default`](Self::with_default), that level
+    /// will be returned. Otherwise `None` will be returned.
+    ///
+    /// A return value of `None` is behaviourly equivalent to [`LevelFilter::OFF`], but
+    /// distinguishes between an explicitly set default level or the default one. If you only care
+    /// about the behaviour you can use `unwrap_or`:
+    ///
+    /// ```
+    /// use tracing_subscriber::filter::{LevelFilter, Targets};
+    ///
+    /// let filter = Targets::new();
+    /// let default_level = filter.default_level().unwrap_or(LevelFilter::OFF);
+    ///
+    /// assert_eq!(default_level, LevelFilter::OFF);
+    /// ```
+    pub fn default_level(&self) -> Option<LevelFilter> {
+        self.0
+            .directives()
+            .into_iter()
+            .find(|d| d.target.is_none())
+            .map(|d| d.level)
```

Comment:
tiny style nit, take it or leave it: this could also be expressed using [`Iterator::find_map`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.find_map)
```suggestion
            .find_map(|d| d.target.is_none().then(d.level))
```

Reply:
Good catch, I did have that at some point along with `bool::then`. Think I got muddled up when checking the MSRV.

Reply:
Oh, I remember now. It's because `bool::then` is only available since 1.50.0 and MSRV is 1.49.0. We can use `filter_map` with an `if`/`else` but that's a bit more verbose.

```rust
            .find_map(|d| {
                if d.target.is_none() {
                    Some(d.level)
                } else {
                    None
                }
            })
```